### ISSUE TITLE
Ungit hanging when handling a lot of (unadded) files

### ISSUE THREAD

**Issue by kekebo** [FredrikNoren/ungit#485]
This maybe overlapping with Issue [#432] ("https://github.com/FredrikNoren/ungit/issues/432) but more specifically it's about handling  *unadded* files.

First I added a whole magento folder (~13.000) files, which instantly led to hanging (on OSX 10.9.5 + Chromium 38.0.2125.122 / Firefox 35.0).
I then started cutting down on added files, but even if I only add a single file, ungit keeps hanging until crash.

I then reinitialised the repo in a subfolder with only a few files, which works fine - so I assume it must be the unadded files causing the hang.

Is there any possibility keeping ungit from scanning those unadded files? (I'm gonna have a look myself - any chance you could point me to the file(s) where this could be happening?)

Love the project, keep it up!

**Comment by kekebo** [comment]
As a temporary workaround for this: Even after adding the whole magento folder, if I edit a single file and commit it before starting ungit, all works fine.

**Comment by FredrikNoren** [comment]
Thanks for the report @kekebo. I believe a solution for #432 would most likely solve this too.

**Comment by codingtwinky** [comment]
btw, if you have a sample git repo with this issue that I can download and test please let me know.  I don't know of any git repo that has this issue.

**Comment by kekebo** [comment]
@codingtwinky 
I tried to reproduce the error with a copy of my local Perl folder, just because it has a lot of files (~20.000)  without being very big (~225MB) - works fine. Magento is around ~13.000 files and 140MB, so file count and file size are unlikely the issue. 
I tested Safari as well with magento - same hang, so it doesn't really seem browser specific either. Maybe it's OSX?
If you want to try reproducing it, just download the current magento version, init a repo, add any amount of files, commit, ungit.

**Comment by kekebo** [comment]
Here's what the Chrome Console logs up to the hang:
```
The key "" is not recognized and ignored. (index):5
App version: 0.8.4 localhost/:32
Event: Object ungit.js:23892
Event: Object ungit.js:23892
Event: Object ungit.js:23892
Uncaught ReferenceError: FlashControl is not defined injected:1
Failed to load resource: net::ERR_CACHE_MISS http://localhost:8448/#/repository?path=%2FUsers%2Frandom%2FDownloads%2Fmagento
GET http://www.gravatar.com/avatar/c6a2bd722115c789661d445bdc68e6b0?default=404 404 (Not Found) c6a2bd722115c789661d445bdc68e6b0:1
Event: Object {event: "init-tooltip"} ungit.js:23892
```

**Comment by codingtwinky** [comment]
I really didn't want to give away my informations to download magento, but I guess my paranoia is groundless...  Anyhow, I did download magento and above error can be reproducible. 

Biggest problem was at `/status` api [call](https://github.com/FredrikNoren/ungit/blob/master/components/staging/staging.js#L108) as one might have imagined.  Although server is handling 25784 files resulting from `/status` call with surprising speed,  creating doms based on object this big would absolutely destroys any browsers.

Honestly, by the time you have 1000+, even 100+, files ungit will be useless.  So instead of optimizing [this](https://github.com/FredrikNoren/ungit/blob/master/components/staging/staging.js#L114) line, I think displaying warning or error message in these extreme scenarios is more appropriate.

Please let me know if anybody disagrees.

**Comment by FredrikNoren** [comment]
@codingtwinky Absolutely agree. I'm wondering if the thing you implemented for diffs, where if there are more than x files it shows a "And 5633 more files" button? Something like that would probably be the best way to handle it

**Comment by kekebo** [comment]
@codingtwinky @FredrikNoren 

Either that or maybe just handling it the way that a commit is handled (showing only 50 files at a time)?
Because the hang is resolved for me once I commit. Although the "show 50 more" function doesn't work on my system for some reason.

**Comment by codingtwinky** [comment]
That bug is addressed at #484 and will be fixed in next release.  And I be implementing the rolling load along with an alert message notifying user.
